### ISSUE TITLE
Do not recommend to disable the daemon on CI

### ISSUE THREAD

**Issue by melix** [gradle/gradle#2824]
It seems that we are still suggesting to disable the daemon on CI: https://docs.gradle.org/current/userguide/gradle_daemon.html#when_should_i_not_use_the_gradle_daemon

We have been using it on CI for months now without any issue, and it's probably a good time to revisit this statement.

### Expected Behavior

Documentation shouldn't recommend to disable the daemon on CI. Note that there still may be cases where using the daemon doesn't make sense, like builds on Travis where a container is spawned for each build: there's no benefit here because the daemon will be killed after each build in any case.

**Comment by Crazyjavahacking** [comment]
Is this really safe in any possible situation?

**Comment by bmuschko** [comment]
@melix Were you planning on working on this for 4.2-rc1?

**Comment by melix** [comment]
Not unless someone else picks this up

Le 30 août 2017 6:40 PM, "Benjamin Muschko" <notifications@github.com> a
écrit :

**Comment by jjohannes** [comment]
>  Note that there still may be cases where using the daemon doesn't make sense, 

I would reword this to _"makes no difference"_. We still start a single-use daemon automatically in many cases even with `--no-daemon`. This has caused confusions several times. So recommending running with `--no-daemon` when we then start a daemon anyway does not make sense in my mind. Hence I would move aways from recommending `--no-daemon` completely.

I vote for removing the complete section _7.6. When should I not use the Gradle Daemon?_. Or replacing it with something like:

> Since Gradle 3.0, the daemon is stable and the recommended mode for running Gradle. We do not recommend disabling it. Note: if you run CI builds in complete isolation, for example containers that kill all running processes at the end of the build, the daemon process will be killed and running with or without daemon effectively makes no difference.

**Comment by pangyikhei** [comment]
I have a big multi-project build and sometimes with the gradle daemon (4.1) enabled, some of the jars created by **gradle build** (some of the projects in the multi-project build are compile Java dependencies of another project in the multi-project build) seem to be held by the gradle daemon process.
Ex. 
I run **gradle clean build** with no daemon started, clean and build tasks complete successfully.
Afterwards when I run **gradle clean build** again, the clean task fails saying it can't delete the jar of the project dependency.
Only after I run **gradle --stop** or kill the process directly, then **gradle clean** completes successfully.

If I disable the daemon, it runs without issue, so I have my CI build (in Visual Studio Team Services running on a On-Premises Build Agent) with the daemon disabled.

**Comment by jjohannes** [comment]
@pangyikhei This sounds like a issue with your build setup independent of where you run the build. If you think this is a bug, please open a separate issue with a reproducible example and if possible a build scan. Otherwise, please ask for support in the forum: https://discuss.gradle.org/c/help-discuss

**Comment by jochenberger** [comment]
It may be a little late to chime in, but I also have a build that is almost not runnable with the daemon enabled. During configure time, the CPU usage stays at about 600% for most several minutes (a lot of it seems to be spent in `CacheAccessWorker`). When disabling the deamon, the configure step finishes in less than a minute.

**Comment by jjohannes** [comment]
@jochenberger This sounds like a general daemon or configuration problem. Unrelated to whether you run it on CI or locally. Can you open a separate issue for this with an example of how to reproduce the issue?

**Comment by jochenberger** [comment]
@jjohannes It was a memory issue. The build in question is quite huge and the daemon was mostly struggling with GC. Raising `Xmx` to 4g helped a lot.
